app/controllers/application_controller.rb

```diff
@@ -5,6 +5,155 @@ class ApplicationController < Sinatra::Base
   configure do
     set :public_folder, 'public'
     set :views, 'app/views'
+    enable :sessions
+    set :session_secret, "fwitter_secret"
   end
 
+  get '/' do
+    #welcome to fwitter!!!
+    erb :index
+  end
+
+  get '/tweets' do
+    #index of all tweets
+    if session[:user_id]
+      @tweets = Tweet.all
+      erb :'tweets/tweets'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/new' do
+    #load form to create a new tweet
+    if session[:user_id]
+      erb :'tweets/create'
+    else
+      redirect to '/login'
+    end
+  end
+
+  post '/tweets' do
+    #creates new tweet
+    if params[:content] == ""
+      redirect to "/tweets/new"
+    else
+      user = User.find_by_id(session[:user_id])
+      @tweet = Tweet.create(:content => params[:content], :user_id => user.id)
+      redirect to "/tweets/#{@tweet.id}"
+    end
+  end
+
+  get '/tweets/:id' do
+    #tweets show
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      erb :'tweets/show'
+    else
+      redirect to '/login'
+    end
+  end
+
+  get '/tweets/:id/edit' do  #load edit form
+    if session[:user_id]
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id == session[:user_id]
```

Comment:
Good job checking that user has access to update tweet before allowing them to edit it. However, we can make a slight improvement on this by finding tweets that are already only associated to the `current_user`, rather than manually checking the `user_id` foreign key of the tweet against the `current_user.id`

```ruby
@tweet = current_user.tweets.find_by(:id => params[:id])
if @tweet
  erb :'tweets/edit'
else
  redirect "/tweets"
end
```

We're using the association collection method provided by [`has_many`](http://api.rubyonrails.org/classes/ActiveRecord/Associations/ClassMethods.html#method-i-has_many) to query the user's tweets. 

Note the usage of [`find_by`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find-by) vs [`find`](http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find). Had we used `find` here, ActiveRecord would raise an error in the event that we could not find a tweet associated with this user. `find_by` will allow this action to fail gracefully, simply returning nil if the user is trying to delete a tweet that they do not have access to.